hnn_core/viz.py

```diff
@@ -242,29 +242,36 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
-    from .dipole import Dipole
+    from .dipole import Dipole, average_dipoles
 
     if ax is None:
         _, ax = plt.subplots(1, 1, constrained_layout=True)
 
     if isinstance(dpl, Dipole):
-        dpl = [dpl]
+        dpls = [dpl]
```

Comment:
No need to rename to `dpls`, changing the `dpl` variable will only persist within the scope of this function.

If the user passes a `Dipole` object into the function, the original variable won't be turned into a `list()`.

Reply:
I think the reason is the following:

"Also, I changed the way to turn dpl into a list, as a function should not modify its parameters especially when the purpose is to draw something. Because the user may expect the dpl to stay the original type after multiple calls of the visualization function (for reasons like refining styles)."

which makes sense to me

Reply:
I see, previously I was using the following form:


```Python
    if isinstance(dpl, Dipole):
        dpl = [dpl]
    else:
        # add average plot only when we have  trials>1.
        # dpl = dpl + [average_dipoles(dpl)]
        dpl.append(average_dipoles(dpl))
```
So this leads to 

<img width="648" alt="image" src="https://user-images.githubusercontent.com/11160442/159516225-70485d41-5c5d-4827-bee5-39ad36413a26.png">

as the `append` modified the `dpl` in an inplace manner. But `dpl = dpl + [average_dipoles(dpl)]` should work so I'll change it back